src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -455,7 +455,8 @@ class GroupConvolution : public ov::pass::MatcherPass {
 
         auto callback = [=](pattern::Matcher& m) {
             auto conv = m.get_match_root();
-            const auto& weights_shape = conv->input_value(1).get_shape();
+            const auto& weight = conv->input_value(1);
+            const auto& weights_shape = weight.get_shape();
```

Comment:
why do we need to write it this way? can we improve API to keep it as a single line w/o warnings?

Reply:
I'll check if possible.
The reason is gcc reporting it is because the `input_value` retruns temporary which is deleted after get shape reference.
In theory this warning is fine but it can be also a false positive as there are many reports about this warning.

There were/are issue it reports false positive `std::span` which is also like `Output<Node>` class a tiny not owning wrapper returned by value. When removed such temporary object compiler don't understand that its life time has no impact on what this reference is pointing.

If there will be not possible to solve in OV code, maybe it would be better to revert change regard dangling reference and for gcc14 this warning should not make compile error.

Reply:
It looks like `ov::Input<T>`, `ov::Output<T>` the member functions which return reference could have attribute [[gnu::no_dangling]]  (added for GCC  >= 14) which should silent this warning and single line would be possible.